app/src/main/java/com/example/android/architecture/blueprints/todoapp/data/source/DefaultTasksRepository.kt

```diff
@@ -97,29 +93,17 @@ class DefaultTasksRepository(
         }
     }
 
-    override suspend fun completeTask(task: Task) {
-        coroutineScope {
-            launch { tasksRemoteDataSource.completeTask(task) }
-            launch { tasksLocalDataSource.completeTask(task) }
-        }
-    }
-
     override suspend fun completeTask(taskId: String) {
-        withContext(ioDispatcher) {
-            getTaskWithId(taskId)?.let { completeTask(it) }
-        }
-    }
-
-    override suspend fun activateTask(task: Task) = withContext<Unit>(ioDispatcher) {
         coroutineScope {
-            launch { tasksRemoteDataSource.activateTask(task) }
-            launch { tasksLocalDataSource.activateTask(task) }
+            launch { tasksRemoteDataSource.completeTask(taskId) }
```

Comment:
Unrelated to your change, but the relationship between the remote and local data sources are a bit unclear here.
Since both are launched run in separate coroutines, either can fail and the failures are independent of each other.

If failing in the remote should not update the local, lets call them both in the same coroutine. If they are truly independent and should run simultaneously, lets be explicit and use [`async`](https://kotlinlang.org/docs/composing-suspending-functions.html#concurrent-using-async)

Reply:
Will address in separate PR